src/unix/fs.c

```diff
@@ -1232,15 +1232,19 @@ static int uv__fs_statx(int fd,
   if (is_lstat)
     flags |= AT_SYMLINK_NOFOLLOW;
 
-  rc = uv__statx(dirfd, path, flags, mode, &statxbuf);
-
-  if (rc == -1) {
+  rc = uv__statx(dirfd, path, flags, mode, &statxbuf); 
+  
+  switch (rc) {
```

Comment:
Can you retain the comment explaining the `rc > 0` case? LGTM apart from that.

Reply:
Thank you, I have added the comment under default.